portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,24 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
 function addRandomFact() {
   const facts =
       ['My favorite color is blue. :)', 'I\'m a big coffee person.', 'I am of Mexican descent.', 'My top genres on Spotify are rock, indie, rap, and Latin!'];
 
-  // Pick a random greeting.
+  // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+async function addGreeting() {
+  const responseFromServer = await fetch('/hello');
+  const textFromResponse = await responseFromServer.text();
+
+  const greetingContainer = document.getElementById('greeting-container');
+  greetingContainer.innerText = textFromResponse;
```

Comment:
I'm being *very* nit-picky here, but I think it's worth pointing out because it gets to the idea of [semantic HTML](https://developer.mozilla.org/en-US/docs/Glossary/Semantics#semantics_in_html). I think you generally don't want to render text as a direct child of a `div`, since `div`s are meant to group related content or provide broad structure to the page. I would probably like to see another tag like a `<p>` nested inside the `<div>` here that you can alter the text of. What you have now should work as intended but could create issues for things like screen readers or web crawlers (for SEO, for example).

That said, the JavaScript here looks great!